ipatests/test_integration/test_installation.py

```diff
@@ -479,3 +483,43 @@ def test_reserved_ip_as_forwarder(self):
         exp_str = ("Invalid IP Address 0.0.0.0: cannot use IANA reserved "
                    "IP address 0.0.0.0")
         assert exp_str in cmd.stdout_text
+
+
+class TestKRAinstallAfterCertRenew(IntegrationTest):
+    """ Test KRA installtion after ca agent cert renewal
+
+    KRA installtion was failing after ca-agent cert gets renewed.
+    This test check if the syptoms no longer exists.
+
+    related ticket: https://pagure.io/freeipa/issue/7288
+    """
+
+    def test_KRA_install_after_cert_renew(self):
+
+        tasks.install_master(self.master)
+
+        # get ca-agent cert contents
+        p12_content = self.master.get_file_contents(paths.DOGTAG_ADMIN_P12)
+
+        # load cert
+        cert = x509.load_der_x509_certificate(p12_content,
+                                              backend=default_backend())
+
+        # get expiry date of agent cert
+        cert_expiry = cert.not_valid_after()
+
+        # move date to grace period so that certs get renewed
+        grace_date = cert_expiry - timedelta(days=20)
+        grace_date = datetime.strftime(grace_date, "%Y-%m-%d %H:%M:%S")
+        self.master.run_command(['date', '-s', grace_date])
+
+        # wait sometime for cert renewal
+        sleep(200)
```

Comment:
This brings non-determinism to the test - how can you be sure that the certificates are renewed after 200 seconds? You need to check that, otherwise you don't know whether you're testing what you actually wanted to test in the first place.

Reply:
@stlaz Could you please provide some hint, how to check it?

Reply:
@stlaz  Sorry, I was confused by github UI. This code is not recent. So I close this thread.

Reply:
@stlaz  More confused then I thought. This is still valid, because IT IS CURRENT code. So, @stlaz  could you please provide any hint here?

Reply:
@celestian I guess a check that the expiration period of the certificates in question changed?

Reply:
@stlaz  good point, @mrizwan93 if you agree, could you please implemented it?